extensions/wikia/Wall/WallThread.class.php

```diff
@@ -77,31 +81,31 @@ private function loadReplyObjs() {
 		}
 	}
 
-	private function loadReplyIdsFromDB($master = false) {
+	/**
+	 * Fetches reply IDs for the thread, using a limit to control large queries
+	 *
+	 * @param DatabaseBase $dbr Database resource
+	 * @param integer $afterId The last reply ID after which the next set is selected
+	 * @return array List of reply IDs
+	 */
+	private function getReplyIdsFromDB( $dbr, $afterId = null ) {
 		// this is a direct way to get IDs
 		// the other one is in Wall.class done in a grouped way
 		// (fetch for many threads at once, set with ->setReplies)
 
-		$title = Title::newFromId( $this->mThreadId );
+		$conditions = array( 'parent_comment_id = '.$this->mThreadId );
 
-		if( empty($title) ) {
-			$title = Title::newFromId( $this->mThreadId, Title::GAID_FOR_UPDATE );
+		if ( (int) $afterId > 0 ) {
+			array_push( $conditions, 'comment_id > '.$afterId );
 		}
 
-		if( empty($title) ) {
-			return ;
-		}
-
-		$dbr = wfGetDB( $master ? DB_MASTER : DB_SLAVE );
-
-		$threadId = $title->getArticleID();
```

Comment:
Yah, ```$title``` was constructed via ```newFromId( ID )``` where ```ID``` is an article ID.  Since its constructed with```$threadId``` we're doing a useless query to return us what we already had to begin with.

The only reason it was even instantiated before was to call ```getDBkey``` and ```getNamespace``` to use as values in the query.  The diff that @mklucsarits pasted above correctly optimizes this out since the ID is the same as a DB key + namespace value.